pkg/koordlet/koordlet.go

```diff
@@ -77,6 +77,22 @@ func NewDaemon(config *config.Configuration) (Daemon, error) {
 	klog.Infof("sysconf: %+v,agentMode:%v", system.Conf, system.AgentMode)
 	klog.Infof("kernel version INFO : %+v", system.HostSystemInfo)
 
+	kubeClient := clientset.NewForConfigOrDie(config.KubeRestConf)
+	crdClient := clientsetbeta1.NewForConfigOrDie(config.KubeRestConf)
+	topologyClient := topologyclientset.NewForConfigOrDie(config.KubeRestConf)
+
+	pleg, err := pleg.NewPLEG(system.Conf.CgroupRootDir)
+	if err != nil {
+		return nil, err
+	}
+
+	metricCache, err := metriccache.NewMetricCache(config.MetricCacheConf)
+	if err != nil {
+		return nil, err
+	}
+
+	statesInformer := statesinformer.NewStatesInformer(config.StatesInformerConf, kubeClient, crdClient, topologyClient, metricCache, pleg, nodeName)
+
```

Comment:
any reason why you moved this block of code forward?

Reply:
getting the kubeletPort from k8s needs to initialize statesInformer in advance
> kubeletPort := int(statesInformer.GetNode().Status.DaemonEndpoints.KubeletEndpoint.Port)

Reply:
that makes sense